Sources/MapboxDirections/DirectionsOptions.swift

```diff
@@ -122,10 +123,133 @@ open class DirectionsOptions: Codable {
 
      - parameter waypoints: An array of `Waypoint` objects representing locations that the route should visit in chronological order. The array should contain at least two waypoints (the source and destination) and at most 25 waypoints. (Some profiles, such as `ProfileIdentifier.automobileAvoidingTraffic`, [may have lower limits](https://docs.mapbox.com/api/navigation/#directions).)
      - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. `ProfileIdentifier.automobile` is used by default.
+     - parameter queryItems: URL query items to be parsed and applied as configuration to the resulting options.
      */
-    required public init(waypoints: [Waypoint], profileIdentifier: ProfileIdentifier? = nil) {
+    required public init(waypoints: [Waypoint], profileIdentifier: ProfileIdentifier? = nil, queryItems: [URLQueryItem]? = nil) {
         self.waypoints = waypoints
         self.profileIdentifier = profileIdentifier ?? .automobile
+        
+        guard let queryItems = queryItems else {
+            return
+        }
+        
+        let mappedQueryItems = Dictionary<String, String>(uniqueKeysWithValues: queryItems.compactMap {
```

Comment:
I would suggest `Dictionary(_:uniquingKeysWith:)` as this one might led to a runtime error if there are duplicates in the query items (which is possible in general). `uniquingKeysWith` might be like `(_, latestValue) in latestValue`

Reply:
Updated. Thanks for the suggestion